#! /bin/sh /usr/share/dpatch/dpatch-run
## 20_extra_colorspace_check.dpatch by  <fpeters@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: added extra check against value read for color
## DP: space (closes: #398426)

@DPATCH@
diff -urNad libexif-0.6.13~/libexif/exif-entry.c libexif-0.6.13/libexif/exif-entry.c
--- libexif-0.6.13~/libexif/exif-entry.c	2005-10-03 20:32:03.000000000 +0200
+++ libexif-0.6.13/libexif/exif-entry.c	2007-05-08 11:37:44.062021561 +0200
@@ -904,7 +904,7 @@
 		/* Find the value */
 		for (j = 0; list2[i].elem[j].values &&
 			    (list2[i].elem[j].index < v_short); j++);
-		if (list2[i].elem[j].index != v_short) {
+		if (list2[i].elem[j].index != v_short || v_short == 65535) {
 			snprintf (val, maxlen, _("Internal error (unknown "
 				  "value %i)."), v_short);
 			break;
